### ISSUE TITLE
The inconsistency checker reports two identical translations as inconsistent

### ISSUE THREAD

**Issue by mudrd8mz** [WeblateOrg/weblate#3494]
**Description**
I am experiencing this in a local environment - test installation of Weblate 3.10.3. I have two components in a project that represent two versions of a software (Moodle plugin). The English original of the string reads "Loading syntax help…" and it has been translated into Czech as "Nahrávám nápovědu k syntaxi…" in both components.

The translations are identical in both components (if it helps - the link to both has the parameter `?checksum=7d8c81a602eaba16` in the URL) and yet the inconsistency checker reports them:

Am I missing something? Is there some additional task to run or so?

**Screenshots**

![image](https://user-images.githubusercontent.com/85831/74661238-d7c2c980-5197-11ea-881a-cb3d591c0139.png)

**Server configuration and status**
* [check_deploy.txt](https://github.com/WeblateOrg/weblate/files/4214210/check_deploy.txt)
* [list_versions.txt](https://github.com/WeblateOrg/weblate/files/4214211/list_versions.txt)

**Comment by nijel** [comment]
Can you please try rerunning checks to see if that helps? In `weblate shell`:

```python
from weblate.trans.models import Project
project = Project.objects.get(name="Marklar editor")
project.run_target_checks()
project.invalidate_stats_deep()
```

**Comment by mudrd8mz** [comment]
```
~/dev/weblate-env $ weblate shell
Python 3.6.9 (default, Nov 17 2019, 16:59:36) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from weblate.trans.models import Project
>>> project = Project.objects.get(name="Marklar editor")
>>> project.run_target_checks()
INFO moodle-editor_marklar: running batch check: inconsistent
INFO moodle-editor_marklar: running batch check: translated
>>> project.invalidate_stats_deep()
INFO moodle-editor_marklar: updating stats caches
```

It did not help though. I ran the webserver after that, the false failing check still reported.

**Comment by nijel** [comment]
Good, at least it's reproducible :-). Can you please share the translation files triggering this?

**Comment by mudrd8mz** [comment]
These are the files as I found them in `site-packages/data/vcs/` repositories for the two versions:

* [files.zip](https://github.com/WeblateOrg/weblate/files/4219267/files.zip)

They are a result of my experiments of importing the following files into weblate:
* https://github.com/mudrd8mz/moodle-editor_marklar/blob/e1e07dd35852a342f766a6b9de976b684e2c277f/lang/en/editor_marklar.php
* https://git.moodle.org/gw?p=moodle-langpacks.git;a=blob;f=cs/editor_marklar.php;h=5509a800ba0dcf88e498058a68fe662c4985ad92;hb=refs/heads/MOODLE_37_STABLE
* https://git.moodle.org/gw?p=moodle-langpacks.git;a=blob;f=cs/editor_marklar.php;h=7604f3a50fae6a7e9d27cab5db2c485650a30212;hb=refs/heads/MOODLE_38_STABLE

and then translating them inside weblate.

Hope this helps.

**Comment by nijel** [comment]
Unfortunately I can't reproduce it even with your files. Maybe you have some not yet committed changes as well?

Anyway going through the code, it might be hash collision. To figure out this, please perform following:

```python
from weblate.trans.views.edit import get_other_units
from weblate.trans.models import Unit
unit = Unit.objects.get(translation__language__code='cs', translation__component__project__name="Marklar editor", translation__component__mame="editor_marklar - 3.7", context="$string['syntaxloading']")
print(len(get_other_units(unit))
```

If it prints 2 then I am wrong and have to look elsewhere, if the number is higher it's a hash collision.

**Comment by nijel** [comment]
This should be fixed by 8a811e67dd9220235186793160ef09dfb2201924.
